### ISSUE TITLE
Problem to build project

### ISSUE THREAD

**Issue by Heixiaomeng** [EthVM/EthVM#545]
* **I'm submitting a ...**
      bug report

* **Feature Request**
   Try to learn a project to realise a blockchain, including front and back. We want to use EthVM for front.

* **Bug Report**
    - What is the current behavior?

    Follow wiki to build the environement.
    - If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem

    I got the projet from the brache of master, after execute "./bin/docker-run.sh up simple", it display an erreur like "ERROR: Volume parity-volume specifies nonexistent driver local-persist". We can't build a project. It like : 
    .....
    Successfully built dbcfd6960bd5
    Successfully tagged ethvm_api:latest
   WARNING: The Docker Engine you're using is running in swarm mode.
   Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.
   To deploy your application across the swarm, use `docker stack deploy`.
   Creating network "ethvm_back" with driver "bridge"
   Creating volume "ethvm_timescale-volume" with default driver
   Creating volume "ethvm_kafka-volume-1" with default driver
   Creating volume "ethvm_zookeeper-volume" with default driver
   Creating volume "ethvm_parity-volume" with local-persist driver
   ERROR: Volume parity-volume specifies nonexistent driver local-persist

Did we miss something? thanks a lot in advance .

**Comment by aldoborrero** [comment]
Hi @Heixiaomeng 

Currently what I recommend you is to start the system from `develop`, `master` branch is too old now and is deprecated, once we release our Beta, we will merge it back to `master`.

Second thing: the wiki has not been updated too much lately, but we will address that soon.

And third:

> I got the projet from the brache of master, after execute "./bin/docker-run.sh up simple", it display an erreur like "ERROR: Volume parity-volume specifies nonexistent driver local-persist". We can't build a project. It like

I've updated docker-compose.yml to don't have enabled the parity volume plugin, so if you try again it should work properly.

**Comment by Heixiaomeng** [comment]
Hi @aldoborrero 
Thanks a lot for your reply, I re-git clone, the project was successfully build.
But there is another small problem. First, my environement is Ubuntu, when I link "http://ethvm.lan" after build, the page shows "Bad Gateway". 
I check some logs like "docker-compose logs -f explorer" and "docker-compose logs -f api", It seems that it has a problem to install yarn, I don't know if this is the cause of the problem. In the logs of api, it shows : 
`WARNING: The DOMAIN variable is not set. Defaulting to a blank string.
WARNING: The PARITY_INSTA_MINING variable is not set. Defaulting to a blank string.
WARNING: The PARITY_VOLUME_MOUNTPOINT variable is not set. Defaulting to a blank string.
WARNING: The PARITY_CHAIN variable is not set. Defaulting to a blank string.
WARNING: The PARITY_MIN_PEERS variable is not set. Defaulting to a blank string.
WARNING: The PARITY_MAX_PEERS variable is not set. Defaulting to a blank string.
WARNING: The TIMESCALE_VERSION variable is not set. Defaulting to a blank string.
WARNING: The CP_VERSION variable is not set. Defaulting to a blank string.
WARNING: The KAFKA_SCHEMA_REGISTRY_URL variable is not set. Defaulting to a blank string.
WARNING: The LENSES_LICENSE_URL variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_USER variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_PASSWORD variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_DB variable is not set. Defaulting to a blank string.
Attaching to ethvm_api_1
api_1                    | yarn install v1.13.0
api_1                    | [1/4] Resolving packages...
api_1                    | [2/4] Fetching packages...
api_1                    | info fsevents@1.2.7: The platform "linux" is incompatible with this module.
api_1                    | info "fsevents@1.2.7" is an optional dependency and failed compatibility check. Excluding it fro                                                                  m installation.
api_1                    | [3/4] Linking dependencies...
api_1                    | warning " > nest-winston@1.0.2" has incorrect peer dependency "@nestjs/common@^5.0.0".
api_1                    | error Could not write file "/var/www/api/yarn-error.log": "EACCES: permission denied, open '/var                                                                  /www/api/yarn-error.log'"
api_1                    | error An unexpected error occurred: "EACCES: permission denied, mkdir '/var/www/api/node_modules                                                                  '".
api_1                    | info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
api_1                    | yarn install v1.13.0
api_1                    | [1/4] Resolving packages...
api_1                    | [2/4] Fetching packages...
api_1                    | info fsevents@1.2.7: The platform "linux" is incompatible with this module.
api_1                    | info "fsevents@1.2.7" is an optional dependency and failed compatibility check. Excluding it fro                                                                  m installation.
api_1                    | [3/4] Linking dependencies...
api_1                    | warning " > nest-winston@1.0.2" has incorrect peer dependency "@nestjs/common@^5.0.0".
api_1                    | error Could not write file "/var/www/api/yarn-error.log": "EACCES: permission denied, open '/var                                                                  /www/api/yarn-error.log'"
api_1                    | error An unexpected error occurred: "EACCES: permission denied, mkdir '/var/www/api/node_modules                                                                  '".
api_1                    | info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
`
And in the log of explorer : 
`WARNING: The DOMAIN variable is not set. Defaulting to a blank string.
WARNING: The PARITY_INSTA_MINING variable is not set. Defaulting to a blank string.
WARNING: The PARITY_VOLUME_MOUNTPOINT variable is not set. Defaulting to a blank string.
WARNING: The PARITY_CHAIN variable is not set. Defaulting to a blank string.
WARNING: The PARITY_MIN_PEERS variable is not set. Defaulting to a blank string.
WARNING: The PARITY_MAX_PEERS variable is not set. Defaulting to a blank string.
WARNING: The TIMESCALE_VERSION variable is not set. Defaulting to a blank string.
WARNING: The CP_VERSION variable is not set. Defaulting to a blank string.
WARNING: The KAFKA_SCHEMA_REGISTRY_URL variable is not set. Defaulting to a blank string.
WARNING: The LENSES_LICENSE_URL variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_USER variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_PASSWORD variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_DB variable is not set. Defaulting to a blank string.
Attaching to ethvm_explorer_1
explorer_1               | yarn install v1.13.0
explorer_1               | [1/5] Validating package.json...
explorer_1               | [2/5] Resolving packages...
explorer_1               | [3/5] Fetching packages...
explorer_1               | info There appears to be trouble with your network connection. Retrying...
explorer_1               | info There appears to be trouble with your network connection. Retrying...
explorer_1               | info There appears to be trouble with your network connection. Retrying...
explorer_1               | info There appears to be trouble with your network connection. Retrying...
explorer_1               | info There appears to be trouble with your network connection. Retrying...
explorer_1               | info There appears to be trouble with your network connection. Retrying...
explorer_1               | info There appears to be trouble with your network connection. Retrying...
explorer_1               | error Could not write file "/var/explorer/yarn-error.log": "EACCES: permission denied, open '/var/explorer/yarn-error.log'"
explorer_1               | error An unexpected error occurred: "https://registry.yarnpkg.com/rxjs/-/rxjs-6.5.1.tgz: ESOCKETTIMEDOUT".
explorer_1               | info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
explorer_1               | yarn install v1.13.0
explorer_1               | [1/5] Validating package.json...
explorer_1               | [2/5] Resolving packages...
explorer_1               | [3/5] Fetching packages...
explorer_1               | info fsevents@1.2.9: The platform "linux" is incompatible with this module.
explorer_1               | info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation.
explorer_1               | [4/5] Linking dependencies...
explorer_1               | warning " > vuex-class@0.3.1" has incorrect peer dependency "vue-class-component@^6.0.0".
explorer_1               | warning "@vue/eslint-config-prettier > eslint-config-prettier@3.3.0" has unmet peer dependency "eslint@>=3.14.1".
explorer_1               | warning "@vue/eslint-config-prettier > eslint-plugin-prettier@3.0.0" has unmet peer dependency "eslint@>= 5.0.0".
explorer_1               | warning "@vue/eslint-config-typescript > typescript-eslint-parser@21.0.2" has unmet peer dependency "eslint@>=4.19.1".
explorer_1               | warning " > babel-core@7.0.0-bridge.0" has unmet peer dependency "@babel/core@^7.0.0-0".
explorer_1               | warning " > eslint-plugin-prettier@3.0.1" has unmet peer dependency "eslint@>= 5.0.0".
explorer_1               | error Could not write file "/var/explorer/yarn-error.log": "EACCES: permission denied, open '/var/explorer/yarn-error.log'"
explorer_1               | error An unexpected error occurred: "EACCES: permission denied, mkdir '/var/explorer/node_modules'".
explorer_1               | info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
explorer_1               | yarn install v1.13.0
explorer_1               | [1/5] Validating package.json...
explorer_1               | [2/5] Resolving packages...
explorer_1               | [3/5] Fetching packages...
explorer_1               | info fsevents@1.2.9: The platform "linux" is incompatible with this module.
explorer_1               | info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation.
explorer_1               | [4/5] Linking dependencies...
explorer_1               | warning " > vuex-class@0.3.1" has incorrect peer dependency "vue-class-component@^6.0.0".
explorer_1               | warning "@vue/eslint-config-prettier > eslint-config-prettier@3.3.0" has unmet peer dependency "eslint@>=3.14.1".
explorer_1               | warning "@vue/eslint-config-prettier > eslint-plugin-prettier@3.0.0" has unmet peer dependency "eslint@>= 5.0.0".
explorer_1               | warning "@vue/eslint-config-typescript > typescript-eslint-parser@21.0.2" has unmet peer dependency "eslint@>=4.19.1".
explorer_1               | warning " > babel-core@7.0.0-bridge.0" has unmet peer dependency "@babel/core@^7.0.0-0".
explorer_1               | warning " > eslint-plugin-prettier@3.0.1" has unmet peer dependency "eslint@>= 5.0.0".
explorer_1               | error Could not write file "/var/explorer/yarn-error.log": "EACCES: permission denied, open '/var/explorer/yarn-error.log'"
explorer_1               | error An unexpected error occurred: "EACCES: permission denied, mkdir '/var/explorer/node_modules'".
explorer_1               | info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
explorer_1               | yarn install v1.13.0
explorer_1               | [1/5] Validating package.json...
explorer_1               | [2/5] Resolving packages...
explorer_1               | [3/5] Fetching packages...
explorer_1               | info fsevents@1.2.9: The platform "linux" is incompatible with this module.
explorer_1               | info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation.
explorer_1               | [4/5] Linking dependencies...
explorer_1               | warning " > vuex-class@0.3.1" has incorrect peer dependency "vue-class-component@^6.0.0".
explorer_1               | warning "@vue/eslint-config-prettier > eslint-config-prettier@3.3.0" has unmet peer dependency "eslint@>=3.14.1".
explorer_1               | warning "@vue/eslint-config-prettier > eslint-plugin-prettier@3.0.0" has unmet peer dependency "eslint@>= 5.0.0".
explorer_1               | warning "@vue/eslint-config-typescript > typescript-eslint-parser@21.0.2" has unmet peer dependency "eslint@>=4.19.1".
explorer_1               | warning " > babel-core@7.0.0-bridge.0" has unmet peer dependency "@babel/core@^7.0.0-0".
explorer_1               | warning " > eslint-plugin-prettier@3.0.1" has unmet peer dependency "eslint@>= 5.0.0".
explorer_1               | error Could not write file "/var/explorer/yarn-error.log": "EACCES: permission denied, open '/var/explorer/yarn-error.log'"
explorer_1               | error An unexpected error occurred: "EACCES: permission denied, mkdir '/var/explorer/node_modules'".
explorer_1               | info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
`
I check also the file ".env". The "DOMAIN=ethvm.lan", I think it was good.
Your projet is so cool, I hope that we can use it in our little projet
Thank you again for your reply, my English is not very good, I hope you can understand ：）
Have a nice day :)

**Comment by aldoborrero** [comment]
@Heixiaomeng how did you started this?

You need to stay at the root folder of the project and execute bin/docker-run.sh up simple. As you can see in the logs the .env file is not being read and necessary variables are not being taken into account (so that's why you're having issues with the different projects)

**Comment by aldoborrero** [comment]
@Heixiaomeng any updates? :)

**Comment by Heixiaomeng** [comment]
@aldoborrero 
Hi Aldo,
Thanks for your messages, I did like you said, and this Monday I retried at the root folder and execute bin/docker-run. sh up simple, as above :
![image](https://user-images.githubusercontent.com/26764465/58970713-371fab80-87ba-11e9-91a5-2287e9d94f4a.png)

But this Monday I had the same problem as before, une page 404 with http://ethvm.lan.

And I just did it again today, it has a new message after execute ./bin/docker-run. sh up simple, as above : 
![image](https://user-images.githubusercontent.com/26764465/58971056-ebb9cd00-87ba-11e9-857c-f10677ca89b0.png)


Thanks a lot for all : )

**Comment by AlexMougenet** [comment]
Hi, i am in the exact same case, and get the exact same error :(
On branch develop, launching a ./bin/docker-run.sh up simple at the root...

**Comment by AlexMougenet** [comment]
Hi
I tried again today on a different Wifi (less limited) and it worked well !
It seems like a curl issue, so the network was the problem in my case

But now i am stuck with an empty EthVM interface, how do i link it with my private blockchain ?

**Comment by aldoborrero** [comment]
@Heixiaomeng looks like that your SQL dataset is not being downloaded correctly, so whenever it tries to import it to the db is corrupted. Can you try again?

As @AlexMougenet has reported it looks like a CURL issue, maybe I need to improve the script to fail fast as soon as the CURL doesn't download a valid file.

@AlexMougenet would you like to move your above message to another issue? So we can tackle both differently. I think that are related but not strictly the same.

Thanks to both

**Comment by Heixiaomeng** [comment]
@aldoborrero 

Hello Aldo,
I retried it for servel times, I have not succeeded till this moment. 
This time is the problem of memoires. I guess that it may be the database. 
I don't know how to free memoires if I didn't start it successfully.

It seems like if it doesn't work, when I restart, the newly downloaded database will not overwrite the previously downloaded database. If we restart often, little by little, the hard disk will have no memories.
 
My steps are ：
- execte bin/docker-run.sh up simple at root ;
- when it does not work, I will execte this command to kill the running docker : 
docker stop $(docker ps -q) && docker rm $(docker ps -a -q) && docker rmi $(docker images -q)
- And after that, I retried with bin/docker-run.sh up simple at root. 
- Sometimes it did not succeed because there is no memories after some attempts. I have to increase physical memory on the virtual machine. I don't know how to delete the old database to free memoires.

Thanks a lot
